### PR TITLE
Feature/add ingredient

### PR DESCRIPTION
**Pull request by anewb87**
#### What's this PR do?
-Successful POST
-Successful GET
-Successful render of get to the page

#### Where should the reviewer start?
scripts.js

#### How should this be manually tested?
npm start
go to pantry
use dropdown to select an ingredient and add an amount, click the button and vioa! 

#### Any background context you want to provide?
Async allowed us to FINALLY get this to work.

We made one function async as well that wasn't making a fetch request and that async function speeded up the rendering time.

### CODE REVIEW

**Review by aliroemhildt — APPROVED** [review]
turns out async is the best thing ever!!! Proud of us for figuring this out!

**Review by kammon10 — APPROVED** [review]
That was impressive work everyone!!

File: src/index.html [hunk 1]
```diff
@@ -164,14 +164,13 @@ <h3>instructions</h3>
...
-      </section>
-      <section class="table">
+      <form class="flex-row center">
+        <label class="form-margin dropdown-label" for="ingredient-options">add new ingredient to pantry:</label>
+        <select class="dropdown form-margin" name="ingredient-options" id="ingredient-options"></select>
+        <input class="quantity-input form-margin" type="number" min="0" placeholder="quantity"/>
+        <button type="button" class="form-margin form-button">add to pantry</button>
+      </form>
+      <section class="table-container">
         <table>
```
Comment (cagallo) [inline src/index.html:174]: smart to put this in a form element and get rid of the section!

File: src/scripts.js [hunk 2]
```diff
@@ -76,10 +80,55 @@ addToCookbookButton.addEventListener('click', toggleCookbookButton);
...
+    console.log(error)
+  }
+}
+
+function reassignUserPantry(data) {
+  const updatedUser = data.find((user) => {
+    return user.id === currentUser.id
+  });
+  currentUser.pantry = new Pantry(updatedUser.pantry)
+}
```
Comment (cagallo) [inline src/scripts.js:131]: needed the async!!

**Review by cagallo — APPROVED** [review]

**Pull request merged**